sympy/solvers/inequalities.py

```diff
@@ -590,6 +590,8 @@ def valid(x):
                     discontinuities))).intersection(
                     Interval(domain.inf, domain.sup,
                     domain.inf not in domain, domain.sup not in domain))
+                if isinstance(critical_points, Intersection):
+                    critical_points = critical_points.args[1]
```

Comment:
Isn't `args[1]` going to be an Interval? How will we iterate over that? Is this line covered?

Reply:
> Isn't args[1] going to be an Interval? How will we iterate over that? Is this line covered?

I think `args[0]` is the interval part.
E.g - 1

https://github.com/sympy/sympy/blob/4766b20cdbb2608099459887eaa7b192f5433e2a/sympy/solvers/tests/test_solveset.py#L2348-L2354

E.g - 2
https://github.com/sympy/sympy/blob/4766b20cdbb2608099459887eaa7b192f5433e2a/sympy/solvers/tests/test_solveset.py#L2175-L2180

Reply:
Here, one more thing is to be noted that `critical_points` actually convert the solution from the `solvify` format to the `solveset` format. (In a vague sense)
Since `solvify` doesn't use to return the solutions for the `IntersectionSet` (from `solveset` in `solvify` function). But after modifying the situation it's giving the solution.
So I think `critical_points` is again converting into `IntersectionSet`. 
Hence this logic for abstracting the solution from `IntersectionSet`.(i.e. critical_points)